Userland/Utilities/cal.cpp

```diff
@@ -16,49 +16,53 @@
 
-// TODO: months are in reality 20-characters wide, but each line contains an extra erronous unneeded space at the end
-// so making this 20 breaks formatting a bit
-static int const month_width = 21;
-static int const year_width = month_width * 3 + strlen("  ") * 2; /* three months plus padding between them */
+static int constexpr month_width = "01 02 03 04 05 06 07"sv.length();
+static int constexpr year_width = 3 * month_width + 2 * "  "sv.length(); /* three months plus padding between them */
 
 static int current_year;
 static int current_month;
 static int current_day;
 
+static ErrorOr<String> join_with_spaces(Vector<String> const& words)
+{
+    StringBuilder sb;
+    TRY(sb.try_join(' ', words));
+    return TRY(sb.to_string());
+}
+
 static ErrorOr<Vector<String>> month_lines_to_print(int month, int year)
 {
     Vector<String> lines;
 
-    // FIXME: Both the month name and month header text should be provided by a locale
-    TRY(lines.try_append(TRY(String::formatted("{:^ {}s}", TRY(String::formatted("{:02} - {:02}", month, year)), month_width))));
+    // FIXME: Both the month name and month header  should be provided by a locale
+    TRY(lines.try_append(TRY(String::formatted("{: ^{}s}", TRY(String::formatted("{:02} - {:02}", month, year)), month_width))));
     TRY(lines.try_append(TRY(String::from_utf8("Su Mo Tu We Th Fr Sa"sv))));
 
-    int day_to_print = 1;
-
     auto date_time = Core::DateTime::create(year, month, 1);
     int first_day_of_week_for_month = date_time.weekday();
-    int days_in_the_month = date_time.days_in_month();
+    int days_in_month = date_time.days_in_month();
 
-    StringBuilder row;
-    for (int i = 1; day_to_print <= days_in_the_month; ++i) {
+    Vector<String> days_in_row;
+    int day = 1;
+    for (int i = 1; day <= days_in_month; ++i) {
         if (i - 1 < first_day_of_week_for_month) {
-            row.append("   "sv);
+            TRY(days_in_row.try_append(TRY(String::from_utf8("  "sv))));
         } else {
-            if (year == current_year && month == current_month && day_to_print == current_day) {
-                row.appendff(INVERT_OUTPUT "{:02}" RESET_OUTPUT " ", day_to_print);
+            if (year == current_year && month == current_month && day == current_day) {
+                TRY(days_in_row.try_append(TRY(String::formatted(INVERT_OUTPUT "{:02}" RESET_OUTPUT, day))));
             } else {
-                row.appendff("{:02} ", day_to_print);
+                TRY(days_in_row.try_append(TRY(String::formatted("{:02}", day))));
             }
-            day_to_print++;
+            day++;
         }
 
         if (i % 7 == 0) {
-            TRY(lines.try_append(TRY(row.to_string())));
-            row.clear();
+            TRY(lines.try_append(TRY(join_with_spaces(days_in_row))));
+            days_in_row.clear();
         }
     }
 
-    TRY(lines.try_append(TRY(row.to_string())));
+    TRY(lines.try_append(TRY(join_with_spaces(days_in_row))));
```

Comment:
The helper seems pointless; there's a join function in new string which can do exactly this in a fallible way.

Reply:
Done